Marlin/src/pins/stm32f4/pins_BLACKPILL_CUSTOM.h

```diff
@@ -0,0 +1,121 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (c) 2022 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (c) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include "env_validate.h"
+
+#ifndef BOARD_INFO_NAME
+  #define BOARD_INFO_NAME "Custom Blackpill"
+#endif
+
+#ifndef DEFAULT_MACHINE_NAME
+  #define DEFAULT_MACHINE_NAME "SLA Printer"
+#endif
+
+#define DISABLE_DEBUG                      false  // DISABLE_(DEBUG|JTAG) is not supported for STM32F4.
+#define ALLOW_STM32F4
+#define BOARD_NO_NATIVE_USB
+
+//
+// EEPROM
+//
+#if NO_EEPROM_SELECTED
+  #define IIC_BL24CXX_EEPROM                      // EEPROM on I2C-0
+  //#define SDCARD_EEPROM_EMULATION
+#endif
+
+//
+// Servos
+//
+#if !defined(SERVO0_PIN) && !defined(HAS_PIN_27_BOARD)
+  #define SERVO0_PIN                        PC13  // BLTouch OUT
+#endif
+
+//
+// Limit Switches
+//
+#define Z_STOP_PIN                          PA15
+#define X_STOP_PIN                          PA5
+#define Y_STOP_PIN                          PA6
+
+//
+// Filament Runout Sensor
+//
+#define FIL_RUNOUT_PIN                      PC15  // "Pulled-high"
+
+//
+// Steppers
+//
+#define X_STEP_PIN                          PC_2
+#define X_DIR_PIN                           PB9
+#define X_ENABLE_PIN                        PC_3  // Shared
```

Comment:
@jmz52 — I notice that using your env suggestion, without our own `variant.h` file this pin and a few others are not defined. I presume that our own `variant.h` file would end up defining these pins, but it has me wondering how all these other pins are defined, but just these few stragglers got left out. What's up with that?

Reply:
PC0...PC12 are not defined as there are no such pins on STM32F401CCU6 MCU.
This is entry line MCU with just 35 IO pins:
```
PA0-PA15
PB0-PB10 and PB12-PB15
PC13-PC15
PH0 and PH1
```
Pins PB11 and PC0-PC12 do not exist and are not defined in either form (i.e. PB11 or PB_11)

Complete list of pins available for F401Cx MCU is defined in one of these files (depends on version of framework-arduinoststm32)
.platformio\packages\framework-arduinoststm32@4.10900.200819\variants\Generic_F401Cx\variant.cpp 
.platformio\packages\framework-arduinoststm32\variants\STM32F4xx\F401CC(F-U-Y)_F401C(B-D-E)(U-Y)\variant_generic.cpp

Reply:
I see. So, if we used a variant that is specific to the CC MCU we should also get complaints that `PC_2`, `PC_3`, etc. don't exist. Then, we'll need to choose different pins for these functions.

Reply:
Hmm, I guess it would be best to get the pins defined with the same pattern (e.g., `PC2` instead of `PC_2`) as the others in our pile. Up to now we have done this by making our own variant, adding the missing pins in the `variant.h` file. (e.g., `buildroot/share/PlatformIO/variants/MARLIN_BIGTREE_BTT002/variant.h`) At some point in the future there may be a published variant in the ststm32 platform with these pins included, and then we can use that official variant instead of having to bundle our own.

Conversely, if all the variants have migrated to the `PC_2` style pattern as a standard, then we ought to consider changing all our pins files and example configurations to conform to that standard.

Reply:
There is an update on mine side. I have working on this from a couple to days and fixed this pin issue. I will be pushing the update to mine forked in couple of days. After do I need to create a new pull request? 
I was thinking Why not we name this board as blackpill_DEFAULT instead to custom. Need your thoughts on this

Reply:
@thinkyhead I have update the pins we can now resolve this conversation.